src/Appwrite/Functions/Validator/Headers.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Appwrite\Functions\Validator;
+
+use Utopia\Validator;
+
+/**
+ * Headers.
+ *
+ * Validates user provided headers
+ */
+class Headers extends Validator
+{
+    protected bool $allowEmpty;
+
+    public function __construct(bool $allowEmpty = true)
+    {
+        $this->allowEmpty = $allowEmpty;
+    }
+
+    /**
+     * Get Description.
+     *
+     * Returns validator description
+     *
+     * @return string
+     */
+    public function getDescription(): string
+    {
+        return 'Invalid header format. Header keys can only contain alphanumeric characters, underscores, and hyphens. Header keys cannot start with "x-appwrite-" prefix.';
+    }
+
+    /**
+     * Is valid.
+     *
+     * @param mixed $value
+     *
+     * @return bool
+     */
+    public function isValid($value): bool
+    {
+        if ($this->allowEmpty && empty($value)) {
+            return true;
+        }
+
+        if (\is_string($value)) {
+            $value = \json_decode($value, true);
+        }
```

Comment:
I think this valudator shouldnt encode. It could lead to validator passing, but then having string value in the action causing trouble.